simpletransformers/language_representation/representation_model.py

```diff
@@ -149,13 +153,13 @@ def encode_sentences(self, text_list, combine_strategy=None, batch_size=32):
                     raise ValueError(
                         "Provided combine_strategy is not valid." "supported values are: 'concat', 'mean' and None."
                     )
-                batch_embeddings = embedding_func(token_vectors).detach().numpy()
+                batch_embeddings = embedding_func(token_vectors).detach() 
             else:
-                batch_embeddings = token_vectors.detach().numpy()
+                batch_embeddings = token_vectors.detach()
             if len(embeddings) == 0:
-                embeddings = batch_embeddings
+                embeddings = batch_embeddings.cpu().detach().numpy()
             else:
-                embeddings = np.concatenate((embeddings, batch_embeddings), axis=0)
```

Comment:
It is more efficient to append the `batch_embeddings` to a list during the loop on batches and then to call the `np.concatenate` only once after the loop ended.